modules/tcfControl.js

```diff
@@ -367,8 +369,19 @@ export function setEnforcementConfig(config) {
   }
 }
 
+export function checkIfCredentialsAllowed(next, options = {}) {
+  if (!options.withCredentials) {
+    next(options);
+    return;
+  }
+  const consentData = gdprDataHandler.getConsentData();
+  options.withCredentials = hasDeviceAccess() && validateRules(ACTIVE_RULES.purpose[1], consentData);
```

Comment:
1. options.withCredentials should not be set unless it starts out as `true` (we don't want to set it if the caller didn't ask for it)
2. I think this would always be false because half of the parameters to `validateRules` are missing. IMO you want to split `validateRules` so that you can have it run only the purpose (and not the vendor) part.
3. If we do override withCredentials due to missing consent we should probably log it (since we don't have the activity system to do it for  us).

Reply:
Yes - I retract point 1